test/bin/scenario.sh

```diff
@@ -196,21 +193,46 @@ launch_vm() {
     fi
 
     echo "Creating ${full_vmname}"
+    local -r vm_wait_timeout=$(( VM_BOOT_TIMEOUT / 60 ))
+
+    # Implement retries on VM creation until the problem is fixed
+    # See https://github.com/virt-manager/virt-manager/issues/498
+    local vm_created=false
+    for attempt in $(seq 5) ; do
+        local vm_create_start
+        vm_create_start=$(date +%s)
+        # FIXME: variable for vcpus?
+        # FIXME: variable for memory?
+        # FIXME: variable for ISO
+        if ! sudo virt-install \
+            --noautoconsole \
+            --name "${full_vmname}" \
+            --vcpus 2 \
+            --memory 4092 \
+            --disk "pool=${VM_STORAGE_POOL},size=20" \
+            --network network="${network_name}",model=virtio \
+            --events on_reboot=restart \
+            --location "${VM_DISK_DIR}/${boot_blueprint}.iso" \
+            --extra-args "inst.ks=${kickstart_url}" \
+            --wait ${vm_wait_timeout} ; then
+
+            # Check if the command exited within 15s due to a failure
+            local vm_create_end
+            vm_create_end=$(date +%s)
+            if [ $(( vm_create_end -  vm_create_start )) -lt 15 ] ; then
+                echo "Error running virt-install on attempt ${attempt}: retrying in 5s"
+                sleep 5
+                continue
+            fi
+            # Stop retrying on timeout error
```

Comment:
How is the timeout induced? There's no longer a `timeout` on line 207.

Reply:
The timeout is induced by the `--wait  ${vm_wait_timeout}` option.

Reply:
Ah, I missed that you added the time argument there.

Reply:
Both me and Patryk saw cases when `virt-install` was not killed by timeout, so we thought it would be better to use the native functionality.